kamon-core/src/main/scala/kamon/trace/ActiveSpan.scala

```diff
@@ -0,0 +1,70 @@
+package kamon.trace
+
+/**
+  * Wraps a [[kamon.trace.Span]] that has been activated in the current Thread. By activated we really mean, it is
+  * stored in a ThreadLocal value inside the tracer until [[kamon.trace.ActiveSpan#deactivate()]] is called.
+  *
+  * When a [[kamon.trace.Span]] is activated it will keep a reference to the previously active Span on the current
+  * Thread, take it's place as the currently active Span and put the original one once this ActiveSpan gets deactivated.
+  *
+  */
+trait ActiveSpan extends Span {
+
+  /**
+    * Sets the currently active Span to whatever Span was active when this Span was activated.
+    *
+    */
+  def deactivate(): Unit
```

Comment:
Nice catch. I think it was just like that in OT and we kept it, but it seems more useful to return the wrapped Span. :+1: